ASM-6815 Add initial Suse 11 and 12 support

aimers1975 opened the pull request:
Preliminary razor tasks for Sles 11 and 12
This will resolve the issue when trying to
create a new razor repo for Sles.

gavin-scott commented on tasks/suse11.task/autoyast.erb:
```diff
@@ -0,0 +1,540 @@
+     partitions.each do |partition|
+       partition.networkObjects.each do |network|
+          interface = interface + "    <interface>\n      <bootproto>dhcp</bootproto>
+      <device>em" + count.to_s + "</device>\n      <name>Broadcom Ethernet controller</name>
```
It's pretty odd to have a multi-line string here along with embedded `\n`'s. I didn't even know you could do a mutli-line string like this in ruby. I would convert this to remove all the `\n`'s and use a heredoc string like this: http://makandracards.com/makandra/1675-using-heredoc-for-prettier-ruby-code

gavin-scott commented on tasks/suse11.task/autoyast.erb:
```diff
@@ -0,0 +1,540 @@
+      <device>em" + count.to_s + "</device>\n      <name>Broadcom Ethernet controller</name>
+      <startmode>auto</startmode>\n      <usercontrol>no</usercontrol>\n    </interface>\n"
+        count += 1
+       end
```
I don't think this loop is doing right thing. It's iterating over each partition that has a PXE network on it and then iterating over each network that is on that partition. So if you had e.g. a 2-port NIC with PXE + workload networks on both port I think this would create configuration for em1 - em4.

Also we don't have to do it right away in this PR but we will need to support static IP configuration here targeted to specific NIC port/partitions.

gavin-scott commented:
The title of this PR doesn't reflect what it does. No need to make the title the same as the jira ticket itself. Title should be something like "ASM-6815 Add support for SuSE 11 and 12"

aimers1975 commented:
@gavin-scott thanks! I'll take care of these, but one thing on the jira ticket name.  6815 is not the jira to add support for suse, its a defect filed because creating an OS repo is broken until the task is in place.  Since this doens't really add the Suse support only creates the initial tasks so creating a repo will work, I thought I'd use that ticket.  If you want me to move this to the ticket for adding support for Suse I can definitely do that, in case it may make it hard to track our Suse support in the future.  The jira for razor support for Suse is 6620.

gavin-scott commented:
@aimers1975 I think the use of that jira is fine, whichever makes more sense to you. I'd just like the title for this PR to be reflective of what it does -- add initial SuSE support.

aimers1975 commented:
@gavin-scott OK, no problem!

aimers1975 commented:
@gavin-scott let me know if you notice anything else.

gavin-scott commented on tasks/suse11.task/autoyast.erb:
```diff
@@ -0,0 +1,544 @@
+        <startmode>auto</startmode>
+        <usercontrol>no</usercontrol>
+      </interface>
+S2
```
This is still a little confusing. How about something like:

       interface = interface + <<-S1
      <interface>
        <bootproto>dhcp</bootproto>
        <device>em#{count}</device>
        <startmode>auto</startmode>
        <usercontrol>no</usercontrol>
      </interface>
      S1

aimers1975 commented:
@gavin-scott Ugh, yes...your way is much better.

aimers1975 commented:
@gavin-scott I added the support for static IPs for the PXE network.  This may actually be working, but I think we need a little more testing on it...Evidently linuxrc command line params only seem to accept interface names, not mac addresses (although they seem at some point to have)...but there is documentation that I found saying that if you don't provide the interface name (which I'm not sure we could figure out in advance) that the installer will find the first available interface that works with the settings.  This seemed to work when I configured a static install interface on my server and switched PXE network between ports 1 and 2.  At the time I only had PXE VLAN enabled on either port 1 or port 2 depending on what I was testing.

gavin-scott commented on tasks/suse11.task/post_install.erb:
```diff
@@ -0,0 +1,55 @@
+<%= render_template("store_ip") %>
+
+#add the puppet master host entry in /etc/hosts
+echo "172.29.15.100 dellasm" >> /etc/hosts
```
This is a bug ;-) Should be `echo "<%= URI.parse(repo_url).host %> dellasm" >> /etc/hosts`
